examples/hello-windows/src/main.rs

```diff
@@ -7,28 +7,28 @@ use winit::{
     window::{Window, WindowId},
 };
 
-struct ViewportDesc {
-    window: Window,
+struct ViewportDesc<'window> {
+    window: &'window Window,
     background: wgpu::Color,
-    surface: wgpu::Surface,
+    surface: wgpu::Surface<'window>,
 }
 
-struct Viewport {
-    desc: ViewportDesc,
+struct Viewport<'window> {
+    desc: ViewportDesc<'window>,
     config: wgpu::SurfaceConfiguration,
 }
 
-impl ViewportDesc {
-    fn new(window: Window, background: wgpu::Color, instance: &wgpu::Instance) -> Self {
-        let surface = unsafe { instance.create_surface(&window) }.unwrap();
+impl<'window> ViewportDesc<'window> {
+    fn new(window: &'window Window, background: wgpu::Color, instance: &wgpu::Instance) -> Self {
+        let surface = instance.create_surface(window).unwrap();
```

Comment:
Some examples continued to work just fine even after we changed `Surface` store the `window`, so I left them as is. Let me know if you want me to change that by just wrapping them in an `Arc`.

Reply:
Let's use Arc as that's the recommended way to deal with this.

Reply:
I left the `hello-triangle` example as is because it didn't require any lifetime introduction. It's also nice to make sure this actually works correctly as well (but that could easily be done in a test instead).

Let me know if you want me to convert the `hello-triangle` example as well.